src/swarm-topography/Swarm/Game/Scenario/Topography/Structure/Recognition/Prep.hs

```diff
@@ -116,3 +115,15 @@ mkEntityLookup nameFunc grids =
       map swap $
         catMaybes $
           zipWith (\idx -> fmap (PositionWithinRow idx r,)) [0 ..] rowMembers
+
+-- * Util
+
+-- | Place the second element of the tuples into bins by
+-- the value of the first element.
+binTuplesHM ::
```

Comment:
How about moving this to `swarm-util`?

Reply:
Let's do so if we get one more usage in the codebase.  As is, having this implementation separate would facilitate splitting the recognizer off to its own package at some point.